site-src/geps/gep-726.md

```diff
@@ -0,0 +1,274 @@
+```
+
+This would be represented with the following API additions:
+```go
+// HTTPRequestRedirect defines a filter that redirects a request. At most one of
+// these filters may be used on a Route rule. This may not be used on the same
+// Route rule as a HTTPRequestRewrite filter.
+type HTTPRequestRedirect struct {
+  // Path defines a path redirect.
+  //
+  // Support: Extended
+  //
+  // +optional
+  Path *HTTPPathRedirect `json:"path,omitempty"`
+  // ...
+}
+
+// HTTPPathRedirectType defines the type of path redirect.
+type HTTPPathRedirectType string
+
+const (
+  // This type of redirect indicates that the complete path will be replaced by
+  // the path redirect value.
+  AbsoluteHTTPPathRedirect HTTPPathRedirectType = "Absolute"
+
+  // This type of redirect indicates that the path match will be replaced by
+  // the path redirect value.
+  PathMatchHTTPPathRedirect HTTPPathRedirectType = "PathMatch"
+)
+
+// HTTPPathRedirect defines configuration for path redirects.
+type HTTPPathRedirect struct {
+  // Type defines the type of path redirect.
+  //
+  // +kubebuilder:validation:Enum=Absolute;PathMatch
+  Type HTTPPathRedirectType `json:"type"`
+
+  // Value defines the HTTP path value to redirect to. An empty value ("")
+  // indicates that the portion of the path to be changed in the redirect should
+  // be removed from the redirect path. For example, a request to "/foo/bar"
+  // with a prefix match of "/foo" would be redirected to "/bar".
+  //
+  // +kubebuilder:validation:MaxLength=1024
+  Value string `json:"value"`
+}
+```
+
+### Rewrites
+
+A new `RequestRewrite` filter can be added to support rewrites. In the following
+example, a a request to `example.com/foo/abc` would be rewritten to
+`example.net/bar/abc`.
+
+```yaml
+kind: HTTPRoute
+apiVersion: gateway.networking.k8s.io/v1alpha2
+metadata:
+  name: http-filter-1
+spec:
+  hostnames:
+  - example.com
+  rules:
+    - matches:
+      - path:
+          type: Prefix
+          value: /foo
+      filters:
+      - type: RequestRewrite
+        requestRewrite:
+          hostname: example.net
+          path:
+            type: ReplacePathMatch
+            value: /bar
+```
+
+This would be represent with the following API additions:
+```go
+// HTTPRequestRewrite defines a filter that modifies a request during
+// forwarding. At most one of these filters may be used on a Route rule. This
+// may not be used on the same Route rule as a HTTPRequestRedirect filter.
+type HTTPRequestRewrite struct {
+  // Hostname is the value to be used to replace the Host header value during
+  // forwarding.
+  //
+  // Support: Extended
+  //
+  // +optional
+  // +kubebuilder:validation:MaxLength=255
+  Hostname *string `json:"hostname,omitempty"`
+
+  // Path defines a path rewrite.
+  //
+  // Support: Extended
+  //
+  // +optional
+  Path *HTTPPathRewrite `json:"path,omitempty"`
+}
+
+// HTTPPathRewriteType defines the type of path rewrite.
+type HTTPPathRewriteType string
```

Comment:
I think having a single path rewrite type would the API more intuitive. Is it possible to combine this and HTTPPathRedirectType types and use the same type in both filters?
I think the cognitive load on the user is going to be high otherwise.

Reply:
To clarify do you mean that Rewrite + Redirect should not overlap in functionality and instead should be used together when both sets of capabilities are needed?

Reply:
No. I mean that HTTPPathRedirectType and HTTPPathRewriteType types should be merged into one.